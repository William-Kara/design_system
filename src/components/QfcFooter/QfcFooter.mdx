import { ArgTypes, Meta } from '@storybook/blocks';
import QfcFooter from '@/components/QfcFooter/QfcFooter.vue';

<Meta title="Modules/Footer" />

# Definition

QfcFooter is placed at the bottom of the page.

## Props

The following props are available for the QfcFooter component:

<ArgTypes of={QfcFooter} />

# Usage

To use QfcFooter, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/bulma/layout/footer.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcFooter.scss';
```

```html
<template>
  <QfcFooter >
</template>
```
