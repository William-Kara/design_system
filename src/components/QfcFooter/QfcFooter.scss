.footer {
  font-size: 1.12rem;
  .footer__logo img {
    width: 52px;
  }
  .footer__link,
  .footer__button {
    display: block;
  }
  .footer__icon {
    display: inline-block;
    margin-right: 1rem;
    margin-left: 1rem;
    margin-top: 2rem;
  }
  .footer__button {
    margin-top: 1.5rem;
  }
}

@include mobile {
  .footer {
    padding-top: 2rem;
    margin-left: 2rem;
    padding-bottom: 5rem;
  }
  .footer .footer__icon {
    margin-left: 0;
    margin-right: 2rem;
  }
}
