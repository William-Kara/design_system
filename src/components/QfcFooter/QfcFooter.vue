<script lang="ts" setup>
import QfcLink from './../QfcLink';
import QfcButton from './../QfcButton';

interface Links {
  title?: string;
  text?: string;
  links: Link[];
}
interface Link {
  label?: string;
  to: string;
  iconSrc?: string;
  isButton?: boolean;
  isExternalLink: boolean;
}
interface Props {
  firstLinks: Links;
  secondLinks: Links;
  thirdLinks: Links;
}
withDefaults(defineProps<Props>(), {
  firstLinks: () => ({
    links: [
      {
        label: 'Mentions légales',
        to: '/mentions-legales',
        isExternalLink: false,
        isButton: false,
      },
      {
        label: 'Politique de confidentialité',
        to: '/politique-de-confidentialite',
        isExternalLink: false,
        isButton: false,
      },
      {
        label: 'Gestion des cookies',
        to: '/gestion-des-cookies',
        isExternalLink: true,
        isButton: false,
      },
    ],
  }),
  secondLinks: () => ({
    title: 'Réseaux sociaux',
    text: 'Suivez-nous pour ne rien manquer !',
    links: [
      {
        to: 'https://twitter.com/',
        iconSrc: '/twitter.svg',
        isExternalLink: true,
        isButton: false,
      },
      {
        to: 'https://fr.linkedin.com/company/',
        iconSrc: '/linkedin.svg',
        isExternalLink: true,
        isButton: false,
      },
    ],
  }),
  thirdLinks: () => ({
    title: 'Contact',
    text: 'Vous avez une question ?',
    links: [
      {
        label: 'Contactez-nous',
        to: '/',
        isExternalLink: false,
        isButton: true,
      },
    ],
  }),
});
</script>
<template>
  <footer class="footer py-6">
    <div class="columns container has-text-left">
      <div class="column footer__column is-2 is-desktop">
        <a class="footer__logo">
          <img src="/image-q.svg" width="168" height="18" alt="logo" />
        </a>
      </div>
      <div class="column footer__column">
        <div v-if="firstLinks && firstLinks.title" class="footer__title h5">
          {{ firstLinks.title }}
        </div>
        <div v-if="firstLinks && firstLinks.text" class="footer__description">
          {{ firstLinks.text }}
        </div>
        <template
          v-for="({ label, to, isExternalLink, iconSrc, isButton }, index) in firstLinks.links"
          :key="index"
        >
          <QfcButton
            class="footer__button"
            v-if="isButton"
            :label="label"
            secondary="true"
            size="small"
          />
          <QfcLink
            v-else
            :class="{
              footer__link: !iconSrc,
              footer__icon: iconSrc,
            }"
            :label="label"
            :href="to"
            :is-external-link="isExternalLink"
            :iconSrc="iconSrc"
          >
          </QfcLink>
        </template>
      </div>
      <div class="column footer__column">
        <div v-if="secondLinks && secondLinks.title" class="footer__title h5">
          {{ secondLinks.title }}
        </div>
        <div v-if="secondLinks && secondLinks.text" class="footer__description">
          {{ secondLinks.text }}
        </div>
        <template
          v-for="({ label, to, isExternalLink, iconSrc, isButton }, index) in secondLinks.links"
          :key="index"
        >
          <QfcButton
            class="footer__button"
            v-if="isButton"
            :label="label"
            secondary="true"
            size="small"
          />
          <QfcLink
            v-else
            :class="{
              footer__link: !iconSrc,
              footer__icon: iconSrc,
            }"
            :label="label"
            :href="to"
            :is-external-link="isExternalLink"
            :iconSrc="iconSrc"
          >
          </QfcLink>
        </template>
      </div>
      <div class="column footer__column">
        <div v-if="thirdLinks && thirdLinks.title" class="footer__title h5">
          {{ thirdLinks.title }}
        </div>
        <div v-if="thirdLinks && thirdLinks.text" class="footer__description">
          {{ thirdLinks.text }}
        </div>
        <template
          v-for="({ label, to, isExternalLink, iconSrc, isButton }, index) in thirdLinks.links"
          :key="index"
        >
          <QfcLink v-if="isButton" :href="to" :is-external-link="isExternalLink" icon-src="">
            <QfcButton class="footer__button" :label="label" secondary="true" size="small"
          /></QfcLink>
          <QfcLink
            v-else
            :class="{
              footer__link: !iconSrc,
              footer__icon: iconSrc,
            }"
            :label="label"
            :href="to"
            :is-external-link="isExternalLink"
            :iconSrc="iconSrc"
          >
          </QfcLink>
        </template>
      </div>
    </div>
  </footer>
</template>

<style lang="scss" scoped src="./QfcFooter.scss" />
