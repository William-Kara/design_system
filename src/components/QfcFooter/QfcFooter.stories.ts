import QfcFooter from './QfcFooter.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcFooter> = {
  component: QfcFooter,
  title: 'Modules/Footer',
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    firstLinks: {
      links: [
        {
          label: 'Mentions légales',
          to: '/',
          isExternalLink: false,
          isButton: false,
        },
        {
          label: 'Politique de confidentialité',
          to: '/',
          isExternalLink: false,
          isButton: false,
        },
        {
          label: 'Gestion des cookies',
          to: '/',
          isExternalLink: false,
          isButton: false,
        },
      ],
    },
    secondLinks: {
      title: 'Réseaux sociaux',
      text: 'Suivez-nous pour ne rien manquer !',
      links: [
        {
          to: '/',
          iconSrc: '/twitter.svg',
          isButton: false,
        },
        {
          to: '/',
          iconSrc: '/linkedin.svg',
          isButton: false,
        },
      ],
    },
    thirdLinks: {
      title: 'Contact',
      text: 'Vous avez une question ?',
      links: [
        {
          label: 'Contactez-nous',
          to: '/',
          isButton: true,
        },
      ],
    },
  },
};
