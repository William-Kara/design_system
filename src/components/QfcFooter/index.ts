import QfcFooter from './QfcFooter.vue';

export { QfcFooter };
export default QfcFooter;
