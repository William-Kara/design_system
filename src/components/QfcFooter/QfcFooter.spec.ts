import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcFooter from './QfcFooter.vue';

let wrapper: any = null;

const propsValues = {
  firstLinks: {
    links: [
      {
        label: 'link 1',
        to: '/link-1',
        isExternalLink: false,
        isButton: false,
      },
      {
        label: 'link 2',
        to: '/link-2',
        isExternalLink: false,
        isButton: false,
      },
      {
        label: 'link 3',
        to: '/link-3',
        isExternalLink: false,
        isButton: false,
      },
    ],
  },
  secondLinks: {
    title: 'RÃ©seaux sociaux',
    text: 'Suivez-nous pour ne rien manquer !',
    links: [
      {
        to: '/',
        iconSrc: '/twitter.svg',
        isButton: false,
      },
      {
        to: '/',
        iconSrc: '/linkedin.svg',
        isButton: false,
      },
    ],
  },
  thirdLinks: {
    title: 'Contact',
    text: 'Vous avez une question ?',
    links: [
      {
        label: 'Contactez-nous',
        to: '/',
        isButton: true,
      },
    ],
  },
};

beforeEach(() => {
  wrapper = shallowMount(QfcFooter, {
    propsData: propsValues,
  });
});

describe('QfcFooter Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
