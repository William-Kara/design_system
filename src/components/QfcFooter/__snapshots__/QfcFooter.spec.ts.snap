// Vitest Snapshot v1

exports[`QfcFooter Component > should render correctly 1`] = `
"<footer data-v-69121660=\\"\\" class=\\"footer py-6\\">
  <div data-v-69121660=\\"\\" class=\\"columns container has-text-left\\">
    <div data-v-69121660=\\"\\" class=\\"column footer__column is-2 is-desktop\\"><a data-v-69121660=\\"\\" class=\\"footer__logo\\"><img data-v-69121660=\\"\\" src=\\"/image-q.svg\\" width=\\"168\\" height=\\"18\\" alt=\\"logo\\"></a></div>
    <div data-v-69121660=\\"\\" class=\\"column footer__column\\">
      <!--v-if-->
      <!--v-if-->
      <qfc-link-stub data-v-69121660=\\"\\" label=\\"link 1\\" href=\\"/link-1\\" isexternallink=\\"false\\" underline=\\"false\\" class=\\"footer__link\\"></qfc-link-stub>
      <qfc-link-stub data-v-69121660=\\"\\" label=\\"link 2\\" href=\\"/link-2\\" isexternallink=\\"false\\" underline=\\"false\\" class=\\"footer__link\\"></qfc-link-stub>
      <qfc-link-stub data-v-69121660=\\"\\" label=\\"link 3\\" href=\\"/link-3\\" isexternallink=\\"false\\" underline=\\"false\\" class=\\"footer__link\\"></qfc-link-stub>
    </div>
    <div data-v-69121660=\\"\\" class=\\"column footer__column\\">
      <div data-v-69121660=\\"\\" class=\\"footer__title h5\\">RÃ©seaux sociaux</div>
      <div data-v-69121660=\\"\\" class=\\"footer__description\\">Suivez-nous pour ne rien manquer !</div>
      <qfc-link-stub data-v-69121660=\\"\\" href=\\"/\\" iconsrc=\\"/twitter.svg\\" isexternallink=\\"false\\" underline=\\"false\\" class=\\"footer__icon\\"></qfc-link-stub>
      <qfc-link-stub data-v-69121660=\\"\\" href=\\"/\\" iconsrc=\\"/linkedin.svg\\" isexternallink=\\"false\\" underline=\\"false\\" class=\\"footer__icon\\"></qfc-link-stub>
    </div>
    <div data-v-69121660=\\"\\" class=\\"column footer__column\\">
      <div data-v-69121660=\\"\\" class=\\"footer__title h5\\">Contact</div>
      <div data-v-69121660=\\"\\" class=\\"footer__description\\">Vous avez une question ?</div>
      <qfc-link-stub data-v-69121660=\\"\\" href=\\"/\\" iconsrc=\\"\\" isexternallink=\\"false\\" underline=\\"false\\"></qfc-link-stub>
    </div>
  </div>
</footer>"
`;
